tests/jax/integration/executor_test.py

```diff
@@ -13,265 +13,130 @@
-
-        Returns:
-            system callback components
-        """
-        components = DesignSpec(
-            **system_init,
-            data_server=mocks.MockOnPolicyDataServer,
-            data_server_adder_signature=ParallelSequenceAdderSignature,
-            parameter_server=DefaultParameterServer,
-            executor_parameter_client=ExecutorParameterClient,
-            trainer_parameter_client=mocks.MockTrainerParameterClient,
-            logger=mocks.MockLogger,
-            **executor,
-            trainer=mocks.MockTrainer,
-            trainer_dataset=mocks.MockTrainerDataset,
-            distributor=mocks.MockDistributor,
-        )
-        return components, {}
-
-
-@pytest.fixture
-def test_executor_parameter_server_system() -> System:
-    """Add description here."""
-    return TestSystemExecutorAndParameterSever()
-
-
-# Skip failing test for now
-@pytest.mark.skip
-def test_executor_parameter_server(
-    test_executor_parameter_server_system: System,
-) -> None:
-    """Test if the parameter server instantiates processes as expected."""
-
-    # Environment.
-    environment_factory = functools.partial(
-        debugging_utils.make_environment,
-        env_name="simple_spread",
-        action_space="discrete",
+    # Observe first and observe
+    assert executor._executor.store.adder._writer.history
+    assert list(executor._executor.store.adder._writer.history.keys()) == [
+        "observations",
+        "start_of_episode",
+        "actions",
+        "rewards",
+        "discounts",
+        "extras",
+    ]
+    assert list(
+        executor._executor.store.adder._writer.history["observations"].keys()
+    ) == ["agent_0", "agent_1", "agent_2"]
+    assert (
+        type(executor._executor.store.adder._writer.history["observations"]["agent_0"])
+        == OLT
     )
 
-    # Networks.
-    network_factory = ippo.make_default_networks
-
-    # Build the system
-    test_executor_parameter_server_system.build(
-        environment_factory=environment_factory,
-        network_factory=network_factory,
-        executor_parameter_update_period=20,
+    assert len(executor._executor.store.adder._writer._column_history) != 0
+
+    # Select actions and select action
+    assert list(executor._executor.store.actions_info.keys()) == [
+        "agent_0",
+        "agent_1",
+        "agent_2",
+    ]
+    assert list(executor._executor.store.policies_info.keys()) == [
+        "agent_0",
+        "agent_1",
+        "agent_2",
+    ]
+    num_possible_actions = environment_factory().action_spec()["agent_0"].num_values
+    assert (
+        lambda: x in range(0, num_possible_actions)
+        for x in list(executor._executor.store.actions_info.values())
+    )
+    assert (
+        lambda: key == "log_prob"
+        for key in executor._executor.store.policies_info.values()
     )
 
+
+def test_executor_single_process_without_adder(test_system_sp: System) -> None:
```

Comment:
Do we need this? The system would always have an adder. If we want to test the evaluator, should we call run_episode on the evaluator process rather?

Reply:
If we want to test the evaluator, shouldn't we just call run_episode on the evaluator process rather?

Reply:
If the running without adder not necessary to be tested I will remove it! Do you think it is unecessary?

Reply:
I made that test because I wanted to check the observe_first and observe because if we don't have an adder they won't update some parameters

Reply:
I think we don't need it but no harm in keeping it in. Lets see what the second reviewer says and we can resolve conversation based on that

Reply:
What are your thoughts on having a method to test evaluator?

Reply:
I think we need to make it seperately what do you think?

Reply:
@DriesSmit  or @RuanJohn can someone of you provide his thought about this?

Reply:
I think best option is make the node an evaluator and keep the test as same (without manually removing adder)

Reply:
does the evaluator will remove the adder from the executor?

Reply:
yeah in builder.py 
self.store.adder = None

Reply:
Oh I see!

Reply:
So should we keep the without adder test or not?

Reply:
sorted by separated evaluator and executor